services/src/main/java/org/keycloak/exportimport/util/ExportUtils.java

```diff
@@ -489,40 +501,45 @@ public static UserRepresentation exportUser(KeycloakSession session, RealmModel
         }
 
         // Role mappings
-        Set<RoleModel> roles = user.getRoleMappings();
-        List<String> realmRoleNames = new ArrayList<>();
-        Map<String, List<String>> clientRoleNames = new HashMap<>();
-        for (RoleModel role : roles) {
-            if (role.getContainer() instanceof RealmModel) {
-                realmRoleNames.add(role.getName());
-            } else {
-                ClientModel client = (ClientModel)role.getContainer();
-                String clientId = client.getClientId();
-                List<String> currentClientRoles = clientRoleNames.get(clientId);
-                if (currentClientRoles == null) {
-                    currentClientRoles = new ArrayList<>();
-                    clientRoleNames.put(clientId, currentClientRoles);
+        if (options.isGroupsAndRolesIncluded()) {
+            Set<RoleModel> roles = user.getRoleMappings();
+            List<String> realmRoleNames = new ArrayList<>();
+            Map<String, List<String>> clientRoleNames = new HashMap<>();
+            for (RoleModel role : roles) {
+                if (role.getContainer() instanceof RealmModel) {
+                    realmRoleNames.add(role.getName());
+                } else {
+                    ClientModel client = (ClientModel)role.getContainer();
+                    String clientId = client.getClientId();
+                    List<String> currentClientRoles = clientRoleNames.get(clientId);
+                    if (currentClientRoles == null) {
+                        currentClientRoles = new ArrayList<>();
+                        clientRoleNames.put(clientId, currentClientRoles);
+                    }
+
+                    currentClientRoles.add(role.getName());
                 }
+            }
 
-                currentClientRoles.add(role.getName());
+            if (realmRoleNames.size() > 0) {
+                userRep.setRealmRoles(realmRoleNames);
+            }
+            if (clientRoleNames.size() > 0) {
+                userRep.setClientRoles(clientRoleNames);
             }
         }
 
-        if (realmRoleNames.size() > 0) {
-            userRep.setRealmRoles(realmRoleNames);
-        }
-        if (clientRoleNames.size() > 0) {
-            userRep.setClientRoles(clientRoleNames);
+        // Credentials - extra security, do not export credentials if service accounts
+        if (internal) {
```

Comment:
Same as above. I think this might be out of scope of this PR and should be discussed separately.

Reply:
More or less the same answer. Users are only exported in the boot process, so internal is always true until now when a user is exported. But this part can be removed if you want. This was just done for extra security, to not export a password never in a web (non-boot) export. In theory a service account never has a password. Tell me if you want me to remove this.